core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -470,10 +477,40 @@ def train_multi_levels(
             model_names_fit += base_model_names + aux_models
         if self.model_best is None and len(model_names_fit) != 0:
             self.model_best = self.get_model_best(can_infer=True, infer_limit=infer_limit, infer_limit_as_child=True)
-        self._time_limit = None
+        self._fit_cleanup()
```

Comment:
This is an interesting idea. I think for now I will keep it as is, but this could be something worth adopting in future. I've made essentially zero use of `yield` in AutoGluon so far, mostly due to my unfamiliarity.